### DISCUSSION TITLE
prompt=none does not redirect to login endpoint

### DISCUSSION THREAD

**Discussion by ludydoo** [ory/hydra#2842] [Q&A] [answered]
Hello all,

I'm working on setting up an identity system for a large humanitarian organization, and we are using hydra to wrap our login server in a oidc-compliant interface. Our login server can be configured to federate identities from multiple upstream identity providers, as well as allowing some users to login with other methods (username, password). This also allows us to use a single identity provider (hydra) for all apps of our organization.

The setup looks like
1. React Frontend App
2. Go Resource Server
3. Go Login Server
4. External IDP

The token exchange happens between the **Resource Server** -> **Hydra** with an `authorization_code` grant. **Hydra** redirects the user to the **Login Server**. There, the user would be prompted to select the **Identity Provider** he/she wishes to log in with (Log in with Github, Log in with ...) . The user would perform the login flow. Upon success, the **Login Server** would receive the tokens from the **Identity Provider**, store them, and accept the Login/Consent request.

The problem arises when wanting to perform silent renewals (`prompt=none`). if the **React App** requests a token renewal, then **Resource Server** will redirect the request to `hydra?prompt=none&...`. At this point, it seems like **Hydra** will *not* forward the request to the **Login Server**. The problem it causes in this situation is, that if the user was logged in using an upstream **Identity Provider**, we would actually like to call the upstream identity provider with `prompt=none` as well, to refresh the user tokens. Effectively acting as a broker. 

While writing this message, I realize that if say, I log in to **MySpace** by doing **Log In with GitHub**, if I log out of **GitHub**, I'm not logged out of **MySpace**. Is this what I'm trying to achieve here ?

But if the user was, for example, entirely removed from **Identity Provider**, the Organization would not want that user to still have access. Would I require something like **SCIM** to immediately invalidate all the sessions of that user, if that user was deleted from the**Identity Provider** ?

**Comment by aeneasr** [comment, marked as answer]
Hey there, great to have you!

> While writing this message, I realize that if say, I log in to MySpace by doing Log In with GitHub, if I log out of GitHub, I'm not logged out of MySpace. Is this what I'm trying to achieve here ?

Exactly! You might however have some luck with https://www.ory.sh/hydra/docs/concepts/logout/

> The problem arises when wanting to perform silent renewals (prompt=none). if the React App requests a token renewal, then Resource Server will redirect the request to hydra?prompt=none&.... At this point, it seems like Hydra will not forward the request to the Login Server. The problem it causes in this situation is, that if the user was logged in using an upstream Identity Provider, we would actually like to call the upstream identity provider with prompt=none as well, to refresh the user tokens. Effectively acting as a broker.

That should be possible, Ory Hydra does indeed redirect to the "login & consent app" even if prompt=none. In that case, the `skip` value of the consent / login request will be `true`, indicating the UI that it should just accept the flow (or reject it if e.g. the user is banned).

Hope this helps!

ps: Is the project you're hinting at Norwegian Refugee Council? :) If so, that'd be awesome!

**Reply by ludydoo** [reply]
Thanks @aeneasr for your reply :) Very helpful. I'll have a look at the logout endpoint.

And yes! This project is part of an open-source initiative by Norwegian Refugee Council in order to (partly) solve the challenge of assessing NRC's beneficiaries identity. If you are curious, you can see the current state of affairs in the link below. We are actively developing this at the moment ;)

https://github.com/nrc-no/core/tree/COR-341-admin-panel/pkg/server/login

**Reply by aeneasr** [reply]
Awesome! Looking forward to hear how you're proceeding :) Also know that we have a free program Ory Cloud for non-profits. Might be useful to you :) We do not support Ory Hydra there yet though, but it's on the road map for 2022!

**Reply by vinckr** [reply]
Hey @ludydoo how has it been going with Ory in the meantime? Feel free to reach out to me if you have any questions also on Ory Cloud or otherwise here or in [the chat](https://slack.ory.sh/).
